Document HttpServerOptionsCustomizer (#34997)
quarkusio/quarkus: docs/src/main/asciidoc/http-reference.adoc

Use `quarkus.http.access-log.exclude-pattern=/some/path/.*` to exclude all entries concerning the path `/some/path/...` (_including subsequent paths_) from the log.
====

== Arbitrary customizations

> Quarkus allows users to arbitrarily customize the options of HTTP server(s) that is(are) started by Quarkus via the use of `io.quarkus.vertx.http.HttpServerOptionsCustomizer`.

Reviewer: ```suggestion
Quarkus allows users to arbitrarily customize the options of HTTP servers started by Quarkus via the use of `io.quarkus.vertx.http.HttpServerOptionsCustomizer`.
```

> For example, if the HTTP port needs to be set programmatically, then the following code could be used:

Reviewer 2: Can be everything from https://vertx.io/docs/apidocs/io/vertx/core/http/HttpServerOptions.html configured or there are some limitations that Quarkus imposes?

Author: What kind of limitations are you thinking of? 

There isn't anything I am personally aware of

Author: That's pretty much implied, no?

Reviewer 3: No limitations, but also... no safety belt...

> For example, the HTTP port needs to be set programmatically, then the following code could be used:

Reviewer 4: ```suggestion
For example, if the HTTP port needs to be set programmatically, then the following code could be used:
```

[source,java]
----
import jakarta.inject.Singleton;
import io.quarkus.vertx.http.HttpServerOptionsCustomizer;

@Singleton <1>
public static class MyCustomizer implements HttpServerOptionsCustomizer {

    @Override
    public void customizeHttpServer(HttpServerOptions options) { <2>
        options.setPort(9998);
    }
}
----

> <1> By making the class a bean, Quarkus will take the customizer into account when it starts the Vert.x servers

Reviewer: ```suggestion
<1> By making the class a managed bean, Quarkus will take the customizer into account when it starts the Vert.x servers
```

> <2> We only care about customizing the HTTP server in this case, so we only override the `customizeHttpServer` method.

Reviewer: ```suggestion
<2> In this case, we only care about customizing the main HTTP server, so we just override the `customizeHttpServer` method.
```
